tools/pythonpkg/src/pyconnection.cpp

```diff
@@ -146,14 +146,14 @@ DuckDBPyConnection *DuckDBPyConnection::Execute(const string &query, py::object
 			auto pending_query = connection->PendingQuery(move(statements[i]));
 			auto res = CompletePendingQuery(*pending_query);
 
-			if (!res->success) {
-				throw std::runtime_error(res->error);
+			if (res->HasError()) {
+				res->ThrowError();
 			}
 		}
 
 		prep = connection->Prepare(move(statements.back()));
-		if (!prep->success) {
-			throw std::runtime_error(prep->error);
+		if (prep->HasError()) {
+			prep->error.ToException();
```

Comment:
```suggestion
			prep->error.ThrowError();
```

Reply:
PreparedStatement does not *entirely* share the same API as BaseQueryResult, in that it does not make `error` private, and therefor does not have GetError, GetExceptionType, ThrowError, SetError

I think it is probably better to unify these API's, but I did not want to bloat this PR even further and cause more compile issues that I would have to CI debug as well 😅

Reply:
Okay, but this line doesn't do anything?